sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/agents/SearchKnowledgeAgentAsyncClient.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search.documents.agents;
+
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.search.documents.agents.implementation.KnowledgeAgentRetrievalClientImpl;
+import com.azure.search.documents.agents.implementation.KnowledgeRetrievalsImpl;
+import com.azure.search.documents.agents.models.KnowledgeAgentRetrievalRequest;
+import com.azure.search.documents.agents.models.KnowledgeAgentRetrievalResponse;
+import com.azure.search.documents.agents.models.RequestOptions;
+import reactor.core.publisher.Mono;
+
+/**
+ * This class provides an asynchronous client for interacting with Azure AI Search Knowledge Agents, enabling retrieval of knowledge and data from various configured backing stores.
+ *
+ * <h2>Overview</h2>
+ * <p>
+ * The {@code SearchKnowledgeAgentAsyncClient} exposes asynchronous APIs for sending retrieval requests to a knowledge agent in Azure AI Search. The agent can aggregate and return relevant data from multiple sources, such as Azure Cognitive Search indexes, vector stores, and other knowledge bases configured in your Azure AI Search instance.
```

Comment:
```suggestion
 * The {@code SearchKnowledgeAgentAsyncClient} exposes asynchronous APIs for sending retrieval requests to a knowledge agent in Azure AI Search. The agent can aggregate and return relevant data from multiple sources, such as Azure AI Search indexes, vector stores, and other knowledge bases configured in your Azure AI Search instance.
```

Not fully certain is this documentation is correct as they can only pull data from an index.